tests/python/gaia-ui-tests/gaiatest/apps/contacts/app.py

```diff
@@ -32,7 +32,7 @@ def launch(self):
                 *self._settings_button_locator))))
 
     def switch_to_contacts_frame(self):
-        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == self.name)
+        self.wait_to_be_displayed()
         self.apps.switch_to_displayed_app()
 
```

Comment:
The contacts app does have app://communications.gaiamobile.org/manifest.webapp as origin, no?
In that case, the name doesn't fit here, so you need to add a manifest_url property.
Same goes for the phone app.